main/channel.c

```diff
@@ -1363,14 +1404,23 @@ struct ast_channel_iterator *ast_channel_iterator_by_exten_new(const char *exten
 {
 	struct ast_channel_iterator *i;
 	char *l_exten = (char *) exten;
-	char *l_context = (char *) context;
+	struct ast_channel_callback_data callback_data = {.search = context, .final_match_result = 0};
 
 	if (!(i = ast_calloc(1, sizeof(*i)))) {
 		return NULL;
 	}
 
+  try_again:
 	i->active_iterator = (void *) ast_channel_callback(ast_channel_by_exten_cb,
-		l_context, l_exten, OBJ_MULTIPLE);
+		&callback_data, l_exten, OBJ_MULTIPLE);
+
+	if (callback_data.final_match_result == CMP_RETRY_NEEDED) {
+		/* We ran into a possible deadlock, try again */
+		ast_debug(3, "ast_channel_iterator_by_exten_new() Avoid deadlock trying to find channel by extension@context: %s@%s\n", exten, context);
```

Comment:
ast_debug includes the function name, no need to repeat it.

This could all be simplified to:

`ast_debug(3, "Avoided deadlock trying to find channel by %@%s\n", exten, context);

Reply:
Fixed